src/app/settings/foods/custom/page.tsx

```diff
@@ -38,12 +41,40 @@ const customFoodsColumns: Array<GridColDef<Food>> = [
     width: 100,
     getActions: ({ id, row: food }: GridRowParams<Food>) => [
       ...(food.accessLevel === "PRIVATE"
-        ? [<EditAction key={id} food={food} />]
+        ? [
+            <ShowLabelAction key={id} food={food} />,
+            <EditAction key={id} food={food} />,
+          ]
         : []),
     ],
   },
 ];
 
+let popupState: PopupState;
```

Comment:
This should be passed as a prop, not stored as a global. React can render asynchronously, so storing state in global variables is problematic.

You can move customFoodColumns inside `ManageCustomFoods()` instead of at the top level:

```ts
  const popupState = usePopupState({
    popupId: "show-nutrition-facts-popup",
    variant: "popper",
  });

  const customFoodsColumns: Array<GridColDef<Food>> = [
    { field: "name", headerName: "Food", flex: 2 },
    { field: "brand", headerName: "Brand", flex: 1 },
    {
      field: "accessLevel",
      headerName: "Type",
      valueFormatter: (accessLevel) =>
        accessLevel === "PRIVATE" ? "Custom" : "Public",
    },
    {
      field: "actions",
      type: "actions",
      headerName: "Actions",
      width: 100,
      getActions: ({ id, row: food }: GridRowParams<Food>) => [
        ...(food.accessLevel === "PRIVATE"
          ? [
              <ShowLabelAction key={id} food={food} popupState={popupState} />,
              <EditAction key={id} food={food} />,
            ]
          : []),
      ],
    },
  ];
```

Reply:
I didn't dare move customFoodsColumns but it makes sense to put both in the component.